airbyte-integrations/connectors/source-bamboo-hr/source_bamboo_hr/manifest.yaml

```diff
@@ -0,0 +1,274 @@
+          - payRate
+          - payRateEffectiveDate
+          - payType
+          - paySchedule
+          - payScheduleId
+          - payFrequency
+          - ssn
+          - sin
+          - standardHoursPerWeek
+          - state
+          - stateCode
+          - status
+          - supervisor
+          - supervisorId
+          - supervisorEId
+          - supervisorEmail
+          - terminationDate
+          - workEmail
+          - workPhone
+          - zipcode
+        description: List of fields to include in custom reports
+    additionalProperties: true
+type: DeclarativeSource
+check:
+  type: CheckStream
+  stream_names:
+    - meta_fields_stream
+streams:
+  - name: employees_directory_stream
+    type: DeclarativeStream
+    retriever:
+      type: SimpleRetriever
+      paginator:
+        type: NoPagination
+      requester:
+        path: employees/directory
+        type: HttpRequester
+        url_base: https://api.bamboohr.com/api/gateway.php/{{ config['subdomain'] }}/v1/
+        http_method: GET
+        authenticator:
+          type: BasicHttpAuthenticator
+          password: ''
+          username: '{{ config[''api_key''] }}'
+        request_headers:
+          accept: application/json
+        request_body_json: {}
+        request_parameters: {}
+      record_selector:
+        type: RecordSelector
+        extractor:
+          type: DpathExtractor
+          field_path:
+            - employees
+    primary_key:
+      - id
+    schema_loader:
+      type: InlineSchemaLoader
+      schema:
+        type: object
+        $schema: http://json-schema.org/schema#
+        properties:
+          id:
+            type: string
+          division:
+            type:
+              - 'null'
+              - string
+          jobTitle:
+            type:
+              - 'null'
+              - string
+          lastName:
+            type: string
+          linkedIn:
+            type:
+              - 'null'
+              - string
+          location:
+            type: string
+          photoUrl:
+            type: string
+          firstName:
+            type: string
+          instagram:
+            type:
+              - 'null'
+              - string
+          workEmail:
+            type:
+              - 'null'
+              - string
+          workPhone:
+            type:
+              - 'null'
+              - string
+          department:
+            type: string
+          supervisor:
+            type: string
+          displayName:
+            type: string
+          mobilePhone:
+            type:
+              - 'null'
+              - string
+          photoUploaded:
+            type: boolean
+          preferredName:
+            type:
+              - 'null'
+              - string
+          canUploadPhoto:
+            type: number
+          workPhoneExtension:
+            type:
+              - 'null'
+              - string
+  - name: custom_reports_stream
```

Comment:
The custom report use a dynamic schema function to build the response.
The stream here is not doing this. You need to build a custom component to achieve that.

Check the [original one](https://github.com/airbytehq/airbyte/blob/690479d221210f9510eef35b30ad85b0dcc6283c/airbyte-integrations/connectors/source-bamboo-hr/source_bamboo_hr/source.py#L76-L88)

Reply:
yes but is there any way to make dynamic schema  in no code